### ISSUE TITLE
grantPublicAccess() not granting access on an already existing bucket created manually

### ISSUE THREAD

**Issue by marcindulak** [aws/aws-cdk#7370]
On https://docs.aws.amazon.com/cdk/api/latest/docs/aws-s3-readme.html#importing-existing-buckets there is an example which modifies access permissions on an already existing bucket, created manually from the AWS web console.

The example below does not modify the access permissions on the bucket, and the generated cdk.out/MyS3.template.json has an empty object.

Using https://github.com/aws/aws-cdk/releases/tag/v1.32.2 with typescript.

```typescript
import cdk = require('@aws-cdk/core');
import s3 = require('@aws-cdk/aws-s3');

class MyS3 extends cdk.Stack {
  constructor(scope: cdk.App, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    const bucket = s3.Bucket.fromBucketName(this, 'bucket', 'my-bucket-random-string');
    bucket.grantPublicAccess();
  }
}

const app = new cdk.App();
new MyS3(app, 'MyS3');
app.synth();
```

---

This is a 📕 documentation issue

**Comment by iliapolo** [comment]
Related to https://github.com/aws/aws-cdk/issues/6548

**Comment by iliapolo** [comment]
The fix for this issue should just be a documentation change in the example. The broader scope and solution will be discussed and resolved as part of #6548
